src/main/scala/viper/silver/parser/ParseAst.scala

```diff
@@ -1239,62 +1282,33 @@ case class PAssume(e: PExp)(val pos: (Position, Position)) extends PStmt
 
 case class PInhale(e: PExp)(val pos: (Position, Position)) extends PStmt
 
-case class PVarAssign(idnuse: PIdnUse, rhs: PExp)(val pos: (Position, Position)) extends PStmt
-
-case class PFieldAssign(fieldAcc: PFieldAccess, rhs: PExp)(val pos: (Position, Position)) extends PStmt
-
-case class PMacroAssign(call: PCall, exp: PExp)(val pos: (Position, Position)) extends PStmt
+/** Can also represent a method call or statement macro with no `:=` when `targets` is empty. */
```

Comment:
It seemed unnecessary to make method calls with no targets a special different PAst node, but then have the typing and translation cases for that node duplicate the code for this PAst node. The intuitive reason is that a method call with no targets (`methodCall(...)`) is just shorthand for `:= methodCall(...)`. Once you have only a single node for both, you then also need to include statement macros, since they're indistinguishable.